org.eclipse.jgit/src/org/eclipse/jgit/util/PGPSign.java

```diff
@@ -0,0 +1,160 @@
+package org.eclipse.jgit.util;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+import org.bouncycastle.bcpg.ArmoredOutputStream;
+import org.bouncycastle.bcpg.BCPGOutputStream;
+import org.bouncycastle.bcpg.HashAlgorithmTags;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.PGPPrivateKey;
+import org.bouncycastle.openpgp.PGPSecretKey;
+import org.bouncycastle.openpgp.PGPSecretKeyRing;
+import org.bouncycastle.openpgp.PGPSecretKeyRingCollection;
+import org.bouncycastle.openpgp.PGPSignature;
+import org.bouncycastle.openpgp.PGPSignatureGenerator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPContentSignerBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder;
+import org.eclipse.jgit.annotations.NonNull;
+
+/**
+ * @author mprabhala
+ * @since 5.2
+ *
+ */
+public class PGPSign {
+
+	private static final char[] HEX_ARRAY = "0123456789ABCDEF".toCharArray(); //$NON-NLS-1$
+
+	private static final Path DEFAULT_SECRET_KEY_PATH = Paths
+			.get(System.getProperty("user.home"), ".gnupg", "secring.gpg");
+
+	private PGPSign() {
+		throw new IllegalAccessError("PGP Utility class"); //$NON-NLS-1$
+	}
+
+	/**
+	 * <p>
+	 * Return the first suitable key for signing in the key ring collection. For
+	 * this case we only expect there to be one key available for signing.
+	 * </p>
+	 * @param signingkey
+	 *
+	 * @return the first suitable PGP secret key found for signing
+	 * @throws IOException
+	 *             on I/O related errors
+	 * @throws PGPException
+	 *             on signing errors
+	 */
+	private static PGPSecretKey findSecretKey(String signingkey)
+			throws IOException, PGPException {
+		InputStream secStream = new BufferedInputStream(
+				Files.newInputStream(DEFAULT_SECRET_KEY_PATH));
+		PGPSecretKeyRingCollection pgpSec = new PGPSecretKeyRingCollection(
+				PGPUtil.getDecoderStream(secStream),
+				new JcaKeyFingerprintCalculator());
+		PGPSecretKey secKey = null;
+
+		@SuppressWarnings("unchecked")
+		Iterator<PGPSecretKeyRing> keyrings = pgpSec.getKeyRings();
+		while (keyrings.hasNext() && secKey == null) {
+			PGPSecretKeyRing keyRing = keyrings.next();
+
+			@SuppressWarnings("unchecked")
+			Iterator<PGPSecretKey> keyIter = keyRing.getSecretKeys();
```

Comment:
If a keyring iterator is called "keyrings", shouldn't a key iterator be called "keys"?